File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    
    for i in range(len(s)):
        if not stack:
            stack.append(s[i])
            continue
        
        if stack:
            if s[i] ==')':
                if stack[-1] =='(':
                    stack.pop()
                else:
                    stack.append(s[i])
            else:
                stack.append(s[i])

    if stack:
        answer=False
    
    
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        Boolean answer = true;
        
        for (char c : s.toCharArray()) {
            if (c == '(') {
                stack.push(c);
            } else { 
                if (!stack.isEmpty()) {
                    stack.pop();
                } else {
                    return false; 
                }
            }
        }
        if(!stack.isEmpty()){
            return false;
        }
        return answer;
    }
}
